crates/chain/src/tx_data_traits.rs

```diff
@@ -64,20 +64,35 @@ impl<A: Anchor> Anchor for &'static A {
 pub trait Append {
     /// Append another object of the same type onto `self`.
     fn append(&mut self, other: Self);
+
+    /// Returns whether the structure is considered empty.
+    fn is_empty(&self) -> bool;
```

Comment:
Let's say I wanted to implement `Append` for a `u32` to represent a monotonically increasing value -- I guess `u32::MIN` would return true for `is_empty`.

Reply:
I think implementing `Append` for `u32` is a rather confusing combination. `Option<u32>` or using a wrapped type makes more sense for me.